aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,170 @@
+
+## Running .NET in the browser
+
+Running .NET code inside web browsers is made possible by a relatively new technology, WebAssembly (abbreviated *wasm*). WebAssembly is based on open web standards and is supported in web browsers without plugins. WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed.
+
+Security isn't a major concern because WebAssembly isn't ordinary assembly code (for example, x86/x64) &mdash; WebAssembly is a new bytecode format that accesses browser functionality with the same capabilities as JavaScript.
+
+Mono is the official .NET runtime for client platforms, such as native mobile apps and games. Recently, the Mono team added support for WebAssembly.
+
+When a Blazor app is built and run in a browser:
+
+1. C# code files and Razor files are compiled into .NET assemblies.
+1. The assemblies and the Mono WebAssembly runtime are downloaded to the browser.
+1. Blazor uses JavaScript to bootstrap the .NET runtime and configures the runtime to load the app's assemblies along with any required base class libraries (BCLs). Document object model (DOM) manipulation and browser API calls are handled via JavaScript interoperability (interop).
+
+## Compilation modes
+
+Mono aims run under WebAssembly in two modes: interpreted and ahead-of-time (AOT).
+
+### Interpreted mode
+
+In interpreted mode, the Mono runtime is compiled into WebAssembly bytecode, but the .NET assembly files aren't compiled into bytecode. The browser can load and execute the Mono runtime, which in turn can load and execute standard .NET assemblies (*\*.dll* files) built by the normal .NET compilation toolchain.
+
+![Interpreted mode](index/_static/interpretedmode.png)
+
+Interpreted mode is similar to how the core internals of the desktop Common Language Runtime (CLR) are distributed precompiled into native code, which then loads and executes .NET assembly files. One key difference is that the desktop CLR uses just-in-time (JIT) compilation extensively to make execution faster, while Mono on WebAssembly is closer to a pure interpretation model.
+
+### Ahead-of-time (AOT) compilation mode
+
+In ahead-of-time (AOT) compilation mode, the app's .NET assemblies are transformed into WebAssembly binaries at build time. At runtime, there's no interpretation. The code executes immediately. It's still necessary to load part of the Mono runtime. Features for low-level .NET services are loaded, such as garbage collection. Features for parsing .NET assemblies aren't required and aren't loaded.
+
+![AOT mode](index/_static/aotmode.png)
+
+This is similar to how the ngen tool has historically allowed AOT compilation of .NET binaries into native machine code. More recently, CoreRT provides a complete native AOT .NET runtime for executing .NET binaries.
```

Comment:
Ngen.exe and link to https://docs.microsoft.com/en-us/dotnet/framework/tools/ngen-exe-native-image-generator